app/src/sharedTest/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt

```diff
@@ -0,0 +1,207 @@
+import org.oppia.android.testing.TestLogReportingModule
+import org.oppia.android.testing.robolectric.RobolectricModule
+import org.oppia.android.testing.threading.TestCoroutineDispatchers
+import org.oppia.android.testing.threading.TestDispatcherModule
+import org.oppia.android.testing.time.FakeOppiaClockModule
+import org.oppia.android.util.accessibility.AccessibilityTestModule
+import org.oppia.android.util.caching.testing.CachingTestModule
+import org.oppia.android.util.gcsresource.GcsResourceModule
+import org.oppia.android.util.logging.LoggerModule
+import org.oppia.android.util.logging.firebase.FirebaseLogUploaderModule
+import org.oppia.android.util.parser.html.HtmlParserEntityTypeModule
+import org.oppia.android.util.parser.image.GlideImageLoaderModule
+import org.oppia.android.util.parser.image.ImageParsingModule
+import org.oppia.android.util.platformparameter.SPLASH_SCREEN_WELCOME_MSG
+import org.oppia.android.util.platformparameter.SPLASH_SCREEN_WELCOME_MSG_VALUE
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Tests to verify the working of Platform Parameter Architecture. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(
+  application = PlatformParameterIntegrationTest.TestApplication::class,
+  qualifiers = "port-xxhdpi"
+)
+class PlatformParameterIntegrationTest {
+
+  @get:Rule
+  val oppiaTestRule = OppiaTestRule()
+
+  @Inject
+  lateinit var testCoroutineDispatchers: TestCoroutineDispatchers
+
+  @Inject
+  lateinit var platformParameterController: PlatformParameterController
+
+  private val mockPlatformParameterList by lazy {
+    val mockSplashScreenWelcomeMsgParam = PlatformParameter.newBuilder()
+      .setName(SPLASH_SCREEN_WELCOME_MSG)
+      .setBoolean(SPLASH_SCREEN_WELCOME_MSG_VALUE)
+      .build()
+
+    listOf<PlatformParameter>(
+      mockSplashScreenWelcomeMsgParam
+    )
+  }
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+    testCoroutineDispatchers.registerIdlingResource()
+  }
+
+  @After
+  fun tearDown() {
+    testCoroutineDispatchers.unregisterIdlingResource()
+  }
+
+  @Test
+  fun testSplashTestActivity_readEmptyDatabase_checkWelcomeMsgIsInvisibleByDefault() {
+    launch(SplashTestActivity::class.java).use {
+      onView(withText(SplashTestActivity.WELCOME_MSG)).check(doesNotExist())
+    }
+  }
+
+  @Test
+  fun testSplashTestActivity_updateEmptyDatabase_readDatabaseValues_checkWelcomeMsgIsVisible() {
+    platformParameterController.updatePlatformParameterDatabase(mockPlatformParameterList)
+    testCoroutineDispatchers.runCurrent()
+
+    launch(SplashTestActivity::class.java).use {
+      onView(withText(SplashTestActivity.WELCOME_MSG))
+        .inRoot(getToastMatcher())
+        .check(matches(isDisplayed()))
+    }
+  }
+
+  // Returns a TypeSafeMatcher which checks for any Toast message in the screen.
```

Comment:
nit: [ ] brackets around TypeSafeMatcher

Reply:
This is not supported as it is not a KDoc. So either we should convert it back KDoc or let it be like this